transport/http/client_test.go

```diff
@@ -45,9 +56,73 @@ func TestHTTPClient(t *testing.T) {
 	case <-time.After(time.Millisecond):
 		t.Fatalf("timeout waiting for %s", headerKey)
 	}
+	// Check that Request Header was successfully received
 	if want := headerVal; want != have {
 		t.Errorf("want %q, have %q", want, have)
 	}
+
+	// Check that the response was successfully decoded
+	response, ok := res.(TestResponse)
+	if !ok {
+		t.Errorf("response should be TestResponse")
+	} else {
+		have = response.String
+		if want := "testbody"; want != have {
+			t.Errorf("want %q, have %q", want, have)
+		}
+
+		// Check that response body was closed
+		b := make([]byte, 1)
+		_, err = response.Body.Read(b)
+		if want := "http: read on closed response body"; err == nil || err.Error() != want {
```

Comment:
Yeah, that's a weird one. Going by the doco, I'd expect it to be [`http.ErrBodyReadAfterClose`](https://golang.org/pkg/net/http/#pkg-variables), but it's an unexported error.
[net/http/transport.go#L1290](https://golang.org/src/net/http/transport.go#L1290)
Still, you're right, that will make the test fragile so it would be better to test that it's not nil or io.EOF.